docs/migration/gcloud/vector.md

```diff
@@ -0,0 +1,162 @@
+
+```bash
+gcloud iam service-accounts create <service_account_name> --display-name "My Service Account"
+```
+
+#### 6. Grant Pub/Sub Editor Role to the New Service Account
+> Skip this step if you have an existing service account.
+
+Grant the `pubsub.editor` role to the service account:
+
+```bash
+gcloud projects add-iam-policy-binding <project_id> --member serviceAccount:<service_account_name>@<project_id>.iam.gserviceaccount.com --role roles/pubsub.editor
+```
+
+#### 7. Generate Key for the New Service Account(Skip if you have an existing one)
+> Skip this step if you have an existing service account key.
+
+If you don't have a service account key, create one:
+
+```bash
+gcloud iam service-accounts keys create ~/key.json --iam-account <service_account_name>@<project_id>.iam.gserviceaccount.com
+```
+#### 8. Grant Pub/Sub Subscriber Role to an Existing Service Account from a Different Project
+> Skip this step if you're not using a service account from a different project
+
+If you have a service account and a key from a different project, you need to add the necessary permissions to the service account in your current project like this: 
+
+```bash
+gcloud projects add-iam-policy-binding <project_id> --member=serviceAccount:<service_account_name>@<existing_project_id>.iam.gserviceaccount.com --role=roles/pubsub.subscriber
+```
+#### 9. Create a Pub/Sub Subscription
+
+Create a Pub/Sub subscription to the topic:
+
+```bash
+gcloud pubsub subscriptions create <subscription_name> --topic <topic_name>
+```
+
+#### 10. Test the Subscription (Optional)
+
+Pull messages from the subscription to test it:
+
+```bash
+gcloud pubsub subscriptions pull --auto-ack <subscription_name>
+```
+---
+
+Remember to replace `<project_id>`, `<logging_service_account>`, `<service_account_name>` and `<existing_project_id>` with your actual project ID and service account names.
+</details>
+
+## 2. Configure Vector
+
+1. When creating your Vector configuration file, ensure the endpoint has a `/splunk` suffix and replace the placeholders as needed.
+2. The `<credentials_path>` should point to your service account key, which would be `key.json` if you followed the above setup steps.
+
+
+###  Sample Configuration
+```yaml
+data_dir: /var/lib/vector
+sources:
+  gcp_pubsub_logs:
+    type: gcp_pubsub
+    project: <project_id>
+    subscription: <subscription_name>
+    credentials_path: key.json # Path to the service account key json
+# Transforms Reference: Transform the data from Sources into desired format
+transforms:
+  remap_file_log:
+    inputs:
+      - 'gcp_pubsub_logs'
+    type: 'remap'
+    source: |
+# The parsed json is stored in the structured variable. The structured variable is merged with the other data/fields.
+      structured = parse_json!(.message)
+      ., err = merge(., structured)
+      if err != null {
+        log("Failed to merge structured data into event: " + to_string(err), level: "error")
+      }
```

Comment:
I'm confused what's happening here. What language is this code? What should a user do if they get this error?